### ISSUE TITLE
rstudio on datahub and r hub gives 404

### ISSUE THREAD

**Issue by felder** [berkeley-dsep-infra/hubs#2585]
rstudio on datahub and r hub gives 404

logs show:

```

[W 2021-08-20 10:09:38.543 SingleUserNotebookApp notebookapp:2034] Error loading server extension jupyter_server_proxy
    Traceback (most recent call last):
      File "/opt/conda/lib/python3.9/site-packages/notebook/notebookapp.py", line 2030, in init_server_extensions
        func(self)
      File "/opt/conda/lib/python3.9/site-packages/jupyter_server_proxy/__init__.py", line 27, in _load_jupyter_server_extension
        server_processes += get_entrypoint_server_processes()
      File "/opt/conda/lib/python3.9/site-packages/jupyter_server_proxy/config.py", line 81, in get_entrypoint_server_processes
        make_server_process(entry_point.name, entry_point.load()())
      File "/opt/conda/lib/python3.9/site-packages/pkg_resources/__init__.py", line 2471, in load
        self.require(*args, **kwargs)
      File "/opt/conda/lib/python3.9/site-packages/pkg_resources/__init__.py", line 2494, in require
        items = working_set.resolve(reqs, env, installer, extras=self.extras)
      File "/opt/conda/lib/python3.9/site-packages/pkg_resources/__init__.py", line 790, in resolve
        raise VersionConflict(dist, req).with_context(dependent_req)
    pkg_resources.ContextualVersionConflict: (nest-asyncio 1.2.0 (/opt/conda/lib/python3.9/site-packages), Requirement.parse('nest-asyncio>=1.5'), {'jupyter-client'})
```

This is surely related to #2582

**Comment by balajialg** [comment]
Thanks @yuvipanda

**Comment by yuvipanda** [comment]
Fixed by https://github.com/berkeley-dsep-infra/datahub/pull/2588

The logs @felder found indicated a version conflict with `jupyter-client`:

```
[W 2021-08-20 10:09:38.543 SingleUserNotebookApp notebookapp:2034] Error loading server extension jupyter_server_proxy
    Traceback (most recent call last):
      File "/opt/conda/lib/python3.9/site-packages/notebook/notebookapp.py", line 2030, in init_server_extensions
        func(self)
      File "/opt/conda/lib/python3.9/site-packages/jupyter_server_proxy/__init__.py", line 27, in _load_jupyter_server_extension
        server_processes += get_entrypoint_server_processes()
      File "/opt/conda/lib/python3.9/site-packages/jupyter_server_proxy/config.py", line 81, in get_entrypoint_server_processes
        make_server_process(entry_point.name, entry_point.load()())
      File "/opt/conda/lib/python3.9/site-packages/pkg_resources/__init__.py", line 2471, in load
        self.require(*args, **kwargs)
      File "/opt/conda/lib/python3.9/site-packages/pkg_resources/__init__.py", line 2494, in require
        items = working_set.resolve(reqs, env, installer, extras=self.extras)
      File "/opt/conda/lib/python3.9/site-packages/pkg_resources/__init__.py", line 790, in resolve
        raise VersionConflict(dist, req).with_context(dependent_req)
    pkg_resources.ContextualVersionConflict: (nest-asyncio 1.2.0 (/opt/conda/lib/python3.9/site-packages), Requirement.parse('nest-asyncio>=1.5'), {'jupyter-client'})
```

pypi.org/project/jupyter-client had just had a release about 18h prior, and it was a major version bump. It was brought in as a transient dependency, and so wasn't pinned. So this was incidentally picked up during the image rebuild of https://github.com/berkeley-dsep-infra/datahub/pull/2581, and caused this problem. Just reverting it won't fix the issue unfortunately, as jupyter-client wasn't pinned and so would just get the latest version anyway. 

 https://github.com/berkeley-dsep-infra/datahub/pull/2588 just pins this <7.0 for the time being so we can investigate what happened.

In the meantime, the last good revision before jupyter-client==7.0 was c1a3143d66aca874e5ca2b626a67cf98b83d432f, so I manually checked it out and deployed with `hubploy deploy datahub hub staging` to test. It worked, so I deployed it for datahub, dlab, ischool, publichealth, r hubs manually to unblock waiting users.
